yaml/stage.go

```diff
@@ -120,3 +122,32 @@ func (s StageSlice) MarshalYAML() (interface{}, error) {
 
 	return output, nil
 }
+
+// MergeEnv takes a list of environment variables and attempts
+// to set them in the stage environment. If the environment
+// variable already exists in the stage, than this will
+// overwrite the existing environment variable.
+func (s *Stage) MergeEnv(environment map[string]string) error {
+	// check if the step container is empty
+	if s == nil || s.Environment == nil {
+		// TODO: evaluate if we should error here
+		//
+		// immediately return and do nothing
+		//
+		// treated as a no-op
+		return nil
+	}
+
+	// check if the environment provided is empty
+	if environment == nil {
+		return fmt.Errorf("empty environment provided for step %s", s.Name)
```

Comment:
I think the error message here has a typo i.e. ~`step`~

```suggestion
		return fmt.Errorf("empty environment provided for stage %s", s.Name)
```